test/Dialect/FIRRTL/canonicalization.fir

```diff
@@ -0,0 +1,19 @@
+; RUN: circt-translate -import-firrtl %s | circt-opt -canonicalize | FileCheck %s
+
+; https://github.com/llvm/circt/issues/847
+; seed: 12
+circuit top_mod :
+  ; CHECK-LABEL: firrtl.module @top_mod
+  module top_mod :
+    input clock: Clock
+    input reset: UInt<1>
+    input inp_h: UInt<0>
+    wire tmp10: SInt<0>
+    wire tmp25: SInt<29>
+    wire tmp28: UInt<1>
+    tmp10 <= asSInt(inp_h)
+    tmp25 <= SInt<29>(-1)
+    tmp28 <= geq(tmp10, tmp25)
+    ; CHECK-NEXT: %tmp10 = firrtl.wire
+    ; CHECK-NEXT: %0 = firrtl.asSInt %inp_h
+    ; CHECK-NEXT: firrtl.connect %tmp10, %0
```

Comment:
I don't know how much utility there is in using the actual fuzzer test case other than to guarantee that this fixes that issue. An alternative, stripped down test case would be:

```scala
circuit ZeroWidthConstFold :
  ; CHECK-LABEL: firrtl.module @ZeroWidthConstFold
  module ZeroWidthConstFold :
    input a: UInt<0>
    output b: UInt<1>
    b <= geq(a, UInt<1>(1))
    ; CHECK:      %[[zero:.+]] = firrtl.constant(1 : ui1)
    ; CHECK-NEXT: firrtl.connect %tmp10, %[[zero]]
```

However, I think the best option is to add tests to `test/Dialect/FIRRTL/canonicalization.mlir` that cover every case/code path that the canonicalization is supposed to hit. (Something like `@ComparisonWithZeroWdith`). What do you think?

Reply:
Fully agreed, covering these cases in `canonicalization.mlir` is a good approach. I wasn't sure whether we do or do not want per-issue test cases. Like buildling a corpus of test cases from all the issues we get to ensure we never regress on something that was filed at one point in the past.

Reply:
Agreed ,the tests should be written in mlir form in canonicalization.mlir, not in a .fir file (because you don't want the parser in the loop).  I don't think you need to exhaustively cover the cases, but exercising at least one of these would be good.

Reply:
Will add this to the mlir file.